File: Array/TwoPointers/removeDuplicate.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int left = 0;
        int n = nums.size();
        for (int right = 1; right < n; ++right) {
            if (nums[left] != nums[right]) {
                left += 1;
                int tmp = nums[left];
                nums[left] = nums[right];
                nums[right] = tmp;
            }
        }
        return left + 1;
    }
};

File: Array/TwoPointers/removeDuplicate.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        left = 0
        n = len(nums)
        if n < 2:
            return n
        for right in range(1, n): 
            if nums[right] != nums[left]:
                left += 1
                nums[left], nums[right] = nums[right], nums[left]
        return left+1